src/redis.ts

```diff
@@ -0,0 +1,1256 @@
+    milliseconds: number,
+    collection: string,
+    options?: string | undefined
+  ) {
+    const command: string = "PEXPIRE";
+    return this._commandParser(command, collection, key, milliseconds, options);
+  }
+
+  pexpireat(
+    key: string,
+    unixTimeMilliseconds: number,
+    collection: string,
+    options?: string | undefined
+  ) {
+    const command: string = "PEXPIREAT";
+    return this._commandParser(
+      command,
+      collection,
+      key,
+      unixTimeMilliseconds,
+      options
+    );
+  }
+
+  pttl(key: string, collection: string) {
+    const command: string = "PTTL";
+    return this._commandParser(command, collection, key);
+  }
+
+  randomkey(collection: string) {
+    const command: string = "RANDOMKEY";
+    return this._commandParser(command, collection);
+  }
+
+  rename(key: string, newKey: string, collection: string) {
+    const command: string = "RENAME";
+    return this._commandParser(command, collection, key, newKey);
+  }
+
+  renamenx(key: string, newKey: string, collection: string) {
+    const command: string = "RENAMENX";
+    return this._commandParser(command, collection, key, newKey);
+  }
+
+  scan(
+    cursor: number,
+    collection: string,
+    pattern?: string | undefined,
+    count?: number | undefined,
+    dataType?: number | undefined
+  ) {
+    const command: string = "SCAN";
+
+    let patternArray: any[] = [];
+    if (pattern !== undefined) {
+      patternArray.push("MATCH");
+      patternArray.push(pattern);
+    }
+
+    let countArray: any[] = [];
+    if (count !== undefined) {
+      countArray.push("COUNT");
+      countArray.push(count);
+    }
+
+    let typeArray: any[] = [];
+    if (dataType !== undefined) {
+      typeArray.push("TYPE");
+      typeArray.push(count);
+    }
+
+    return this._commandParser(
+      command,
+      collection,
+      cursor,
+      ...patternArray,
+      ...countArray,
+      ...typeArray
+    );
+  }
+
+  ttl(key: string, collection: string) {
+    const command: string = "TTL";
+    return this._commandParser(command, collection, key);
+  }
+
+  type(key: string, collection: string) {
+    const command: string = "TYPE";
+    return this._commandParser(command, collection, key);
+  }
+
+  unlink(keys: string[], collection: string) {
+    const command: string = "UNLINK";
+    return this._commandParser(command, collection, ...keys);
+  }
+  // End of GENERIC commands
+
+  // Start of SERVER commands
+  echo(message: string, collection: string) {
+    const command: string = "ECHO";
+    return this._commandParser(command, collection, message);
+  }
+
+  ping(collection: string, message?: string | undefined) {
+    const command: string = "PING";
+    return this._commandParser(command, collection, message);
+  }
+
+  dbsize(collection: string) {
+    const command: string = "DBSIZE";
+    return this._commandParser(command, collection);
+  }
+
+  flushdb(collection: string, asyncFlush?: boolean | undefined) {
```

Comment:
any particular reason why we are not following the naming conventions for function names?

Reply:
Yes, as you can see, all the names are the names of redis commands. Please check the supported commands [Supported commands](https://macrometa.com/docs/queryworkers/redis/)
Developers that work with redis are used to this naming and to give them intuitive developer experience we created method signature that you saw in tests, first we have redis class that differentiate methods from other features of the platform and after that we have redis command name `const response = await c8Client.redis.<rediscommand>(parameters);`
What do you think about this approach?
What is usual naming convention that we follow at Macrometa?

Reply:
@dlozina-macrometa IMO let's say the method is written to Get hash which is `hget` we should name it `getHash` in our jsc8 instead. It is more user-friendly.

@elof @lohaniabhishek We would like to know your thoughts on this.

Reply:
@VikasPatel In that case, we are renaming already well known operations/methods. That will be good for all new people that are getting into redis + macrometa way of doing things, but not people that are already onboarded on redis. This feature or interface will be more of a use for people that are already redis users and need to use benefits of GDN.
I have used this SDK as a good example of how to do it [SDK example](https://docs.upstash.com/redis/sdks/javascriptsdk/getstarted) @elof @lohaniabhishek

Reply:
@elof agreed with method names that we already have so I will close this conversation.